### ISSUE TITLE
Short user names / group names are not found as recipients

### ISSUE THREAD

**Issue by jesmrec** [owncloud/ios-app#569]
### Steps to reproduce
1.  In `config.php` set  `user.search_min_length' => 4`,
2. Create a user called `123` and a group called `456`
3. Try to share an item with them 

### Actual behaviour
Not posible, so they do not reach the 4 characters needed to start searching

### Expected behaviour
They should be posible to share with.

After checking the web UI and desktop client, they follow the value of `user.search_min_length` but in case there is a shorter-name recipient that matches the typed pattern, they show it. This is, repeating the steps 1. and 2. above:

3. Type `123`  (or `456`)

Current: `123` (or `456`) is shown in the list, even when its length does not reach the minimum in `config.php`.

iPhone 6S Plus, iOS 13.1
oC 10.3

**Comment by michaelstingl** [comment]
@pmaier1 @hodyroff should 3-character users should show up if `user.search_min_length' => 4` ? I'd say no (works as designed)

**Comment by jesmrec** [comment]
this one came up to my mind during the review of the autocompletion scenarios in product repo:

https://github.com/owncloud/product/blob/da45996867777adead4fc6d1f18f1221fc068b8c/modules/ROOT/examples/acceptanceTests/features/collaborators/findingCollaborators.feature#L79

**Comment by pmaier1** [comment]
> @pmaier1 @hodyroff should 3-character users should show up if user.search_min_length' => 4 ? I'd say no (works as designed)

In the web UI they show up for exact matches. This should be the case on all platforms, I think.

**Comment by hodyroff** [comment]
Yes, thats my expectation as well, the restriction is for searches, not for exact matches.
Wondering again if there is no standard for stuff like this ...

**Comment by michaelstingl** [comment]
My assumption was, users need to hit ENTER or SEARCH after entering a 3-character username, to fire the search request. We shouldn't fire autocomplete-search requests after the 1st, 2nd or 3rd keystroke?

**Comment by hodyroff** [comment]
Actual behaviour says: "not possible". If that is possible after hitting something is a UX question and should also be the same across clients if not totally unusal on a particular client ...

**Comment by jesmrec** [comment]
That's it, there is no way, no key, nothing to do in order to display the result with the 3-letter-name if the capability is 4. In both mobile clients (i created also issue in android repo).

but also i got @michaelstingl 's point. If the sys-admin set the capability as '4' , the sys-admin must assure there is no usernames or groups which length is shorter than 4.  In that case, the product scenario mentioned above for testing will make no sense.

**Comment by pmaier1** [comment]
> If the sys-admin set the capability as '4' , the sys-admin must assure there is no usernames or groups which length is shorter than 4.

I disagree. Would not want to put this limitation in the product.

> Otherwise it makes no sense for clients to respect this capability?

Yeah, true. I guess the client should just use what the API provides :man_shrugging: 

Generally, the main intent of that option is to make it harder to extract user information via API calls. Controlling the number of requests from clients was actually an interesting side-idea, I guess.

**Comment by jesmrec** [comment]
That means, must clients not attend the capability, and showing all matches with the minimum of 3 characters?

**Comment by davigonz** [comment]
> My assumption was, users need to hit ENTER or SEARCH after entering a 3-character username, to fire the search request. We shouldn't fire autocomplete-search requests after the 1st, 2nd or 3rd keystroke?

> Otherwise it makes no sense for clients to respect this capability?

I have the same question as well, we fire requests to get users and groups once the `search_min_length` capability value is reached. If we also have to show users and groups with less characters than what `search_min_length` capability says, we need to fire requests without taking into account the capability, so it does not make any sense to keep it.

**Comment by pmaier1** [comment]
>  we need to fire requests without taking into account the capability, so it does not make any sense to keep it.

I think that's the way to go if @michaelstingl agrees.

**Comment by davigonz** [comment]
I've just found this note in the [ownCloud User Manual](https://doc.owncloud.com/server/user_manual/files/webgui/sharing.html#creating-shares):

![Screenshot 2019-12-17 at 13 06 15](https://user-images.githubusercontent.com/15261363/70994217-05bf5b00-20ce-11ea-8850-e08caff78071.png)

**"Consequently, search terms shorter than the defined number of characters will not return a list of matches"**

Not sure when was this information published, but if that's the expected behaviour and the users are aware of it, we could close this issue and https://github.com/owncloud/android/issues/2747 since it doesn't need to be fixed.

**Comment by hodyroff** [comment]
Please read the next sentence as well! It says the opposite for exact matches and thats how it shall work.

**Comment by michaelstingl** [comment]
@jesmrec clients should ignore the capability and fire every request after every keystroke to the server (maybe min.delay). Server will decide if there are results…

**Comment by jesmrec** [comment]
Will include in next releases. 1.4 is ok?

**Comment by davigonz** [comment]
> Please read the next sentence as well! It says the opposite for exact matches and thats how it shall work.

Sorry, I misread that, I read it while preparing the 2.14 release for Android in which we support that capability, but now it seems we have to stop using it in coming versions. I hope users updating to 2.14 will not bother with this change in next versions. Thanks
